ext/RMagick/extconf.rb

```diff
@@ -123,42 +123,82 @@ def exit_failure(msg)
       exit(1)
     end
 
+    def detect_imagemagick_package(packages)
+      packages.each do |package|
+        return package if system "pkg-config --exists #{package}"
+      end
+      nil
+    end
+
+    def detect_im6_package
+      packages = %w[
+        ImageMagick-6.Q8
+        ImageMagick-6.Q16
+        ImageMagick-6.Q32
+        ImageMagick-6.Q64
+        ImageMagick-6.Q8HDRI
+        ImageMagick-6.Q16HDRI
+        ImageMagick-6.Q32HDRI
+        ImageMagick-6.Q64HDRI
```

Comment:
The rules for package names follow the description below.
```
MAGICK_ABI_SUFFIX="Q${QUANTUM_DEPTH}"
if test "$enable_hdri" = 'yes'; then
  MAGICK_ABI_SUFFIX="Q${QUANTUM_DEPTH}${MAGICK_HDRI}"
fi
```

Ref. https://github.com/ImageMagick/ImageMagick/blob/main/configure

Reply:
Wondering if we should think about the order of this list? Is this in the same order as it was before?

Reply:
If ImageMagick does not generate multiple package files at once, I think it does not matter in which order they are examined.

Reply:
At least I could not create ImageMagick-6.Q16HDRI and ImageMagick-6.Q16 at the same time.

Reply:
There will be cases where multiple installations will be installed with different configurations on multiple installation paths, so I fixed it at https://github.com/rmagick/rmagick/pull/1359/commits/9b5edaa2cfcc98bb2207caeec36ca1ff14768881